api/base/middleware.py

```diff
@@ -410,3 +410,12 @@ def set_sloan_cookie(self, name: str, value, url, request, resp, custom_domain=N
         # Browsers won't allow use to use these cookie attributes unless you're sending the data over https.
         resp.cookies[name]['secure'] = True
         resp.cookies[name]['samesite'] = 'None'
+
+    def check_browseable_api(self, request):
+        """
+        Should return true if browseable api is requested.
+        :param request:
+        :return:
+        """
+        return request.environ.get('HTTP_ACCEPT') == 'text/html' or \
```

Comment:
as an aside, do you know where `request.environ` comes from? i can't find any mention of it it the django or DRF docs -- django has a [`request.META`](https://docs.djangoproject.com/en/1.11/ref/request-response/#django.http.HttpRequest.META) that seems the same, but i'm confused by `environ`.

also, looks like [DRF's `request.accepted_media_type`](https://www.django-rest-framework.org/api-guide/requests/#accepted_media_type) might be the best way to check the accept header, since it's post-content-negotiation and [will reliably `== 'text/html'` for the browsable api](https://github.com/encode/django-rest-framework/blob/3.8.2/rest_framework/renderers.py#L388)

Reply:
I believe `environ` goes as deep as OS level based on a vague memory of some old bugs I fixed involving it. I bet MattF knows, I'm going to ask him.

Yes, request.accepted_media_type == text/html' much better.

Reply:
`request.environ` is a Common Gateway Interface thing -- an webserver-type framework (e.g. Flask, Werkzeug, Django) that plays nice with a WSGI will have it on request objects.

https://www.python.org/dev/peps/pep-0333/#environ-variables

Reply:
ohh nice, good to know, thanks -- mildly annoyed that django documents the base class `HttpRequest` instead of `WSGIRequest` (also that those are different classes?), but i guess that annoyance will help me remember.